### DISCUSSION TITLE
Bonsai configurations of spinnaker capture

### DISCUSSION THREAD

**Discussion by NofarBart** [bonsai-rx/bonsai#1800] [Q&A] [answered]
Hi! :)
We record videos using spinnaker (we have the bonsai spinnaker library) and would like to know where to set the following features:
1. resolution- we would like to reduce the resolution to 640x480.
2. frames per second rate- we would like to increase this parameter to 150.
3. compression- we would like to compress the videos so the file will be smaller in size, and that the compression will interact with spinnaker camera.

Would appreciate a way without attaching code (by definitions of operators\ workflow only)
Attaching the workflow that we are working with:
![WhatsApp Image 2024-05-16 at 04 19 45](https://github.com/bonsai-rx/bonsai/assets/83591641/7167b8c6-9da8-4c4c-b0f7-a7c4de630ac5)
![WhatsApp Image 2024-05-16 at 04 19 45 (1)](https://github.com/bonsai-rx/bonsai/assets/83591641/98686191-96a9-4ada-97cb-2e7b2c499d58).
We would like to do this in the camera properties itself and not through the VideoWriter. 
Thank in advance,
Nofar

**Comment by bruno-f-cruz** [comment]
1- You should set this via the SpinView software
2- Same as 1- if acquiring in non-trigger mode

Note on 1/2, you can set them through bonsai but making your own custom operator. See example here: https://github.com/SainsburyWellcomeCentre/aeon_acquisition/blob/main/src/Aeon.Acquisition/SpinnakerCapture.cs

3- If your camera has any sort of on-chip compression available, you should be able to set it via the previous two strategies, if not, you need to set it in the VideoWriter node by either specifying a different FourCC, or using VideoWritter from the Ffmpeg package and specifying your video encoding parameters.

Hope it helps!

**Reply by NofarBart** [reply]
It helped! Thank you.

We would like to change the files into .mp4 because even with compression with the desired fps (150) it is recording 2GB in 20 minutes (with DVIX and Area in the ResizeInterpol). Our goal is to get the smallest file in size as we can without impairing rhe resulution.
 
We looked at post #919 and downloaded the FFMPEG_example.zip, we also installed the needed package and we get this error:
![WhatsApp Image 2024-05-20 at 13 21 00](https://github.com/bonsai-rx/bonsai/assets/83591641/43892646-a06d-4499-9867-66155840f339)
These are the parameters:
![WhatsApp Image 2024-05-20 at 13 21 01](https://github.com/bonsai-rx/bonsai/assets/83591641/4e4cb1e5-bc25-4755-99cf-6f461d67f319)
What is the problem?
In addition this is the previous version of VideoWriter we used:
![WhatsApp Image 2024-05-20 at 13 21 01 (1)](https://github.com/bonsai-rx/bonsai/assets/83591641/25673ffd-aa29-4555-809e-8f139b3b4b32)

Nofar

**Reply by bruno-f-cruz** [reply]
You might want to search the group for similar questions. Eg: https://github.com/bonsai-rx/bonsai/discussions/1778#discussioncomment-9333065

**Reply by NofarBart** [reply]
Hi bruno,
We looked at similar questions as #919 #1778 and #1243 but couldn't fix the problem. We would like to save the files as mp4 and while trying to record using FFMPEG package for VideoWriter it says "The system can not find the file spetified". (Even with the parameters used in previous discussions). The transformation to mp4 is crucial for us, would appreciate any advice.
Thanks for everything!
Nofar

**Reply by bruno-f-cruz** [reply]
Where do you have the ffmpeg executable stored?

**Reply by NofarBart** [reply]
Hi, sorry for the delay...
We have it in extentions:
![WhatsApp Image 2024-05-26 at 12 22 54](https://github.com/bonsai-rx/bonsai/assets/83591641/2af11f6a-7c73-49c6-9261-50168545d45d)
![WhatsApp Image 2024-05-26 at 12 22 53](https://github.com/bonsai-rx/bonsai/assets/83591641/365d431c-da19-454b-9a75-9c0731d4cb5b)

**Reply by bruno-f-cruz** [reply, marked as answer]
I don't see a ffmpeg.exe in the extensions folder.

**Reply by NofarBart** [reply]
I tried to download the ffmpeg like this:
![IMG_4071](https://github.com/bonsai-rx/bonsai/assets/83591641/644d0866-628c-4eaa-9162-05769267bec1)
It downloaded a ffmpeg-7.0.1 folder without an exe file inside it- 
![IMG_4072](https://github.com/bonsai-rx/bonsai/assets/83591641/7c9b4df4-e8be-4975-a14b-431f39aede5e)
We still get the same error 🥺
What do I need to change?

**Reply by NofarBart** [reply]
Hi bruno 😄 
The mp4 recording is working! used this video to find the missing .exe and build https://youtu.be/r1AtmY-RMyQ?si=d09p6qor2DsBfMxs
Thank you very much for your help!
Nofar
